profiler/src/ProfilerEngine/Datadog.Profiler.Native/CorProfilerCallback.cpp

```diff
@@ -259,6 +268,8 @@ void CorProfilerCallback::DisposeInternal(void)
         std::this_thread::sleep_for(EngineShutdownSafetyPeriodMS);
 
         Log::Debug("CorProfilerCallback::DisposeInternal():  Pause completed.");
+
+        _this = nullptr;
```

Comment:
Should be moved outside of the if because _this is set unconditionally in the constructor

Reply:
✔️